docs/book/src/cronjob-tutorial/pprof-tutorial.md

```diff
@@ -0,0 +1,61 @@
+# Using Pprof for Controller Profiling
+
+Pprof is a Go based library designed for profiling the binary in order to gather CPU, and Memory related data. Pprof is helpful in finding any sort of bottlenecks that are appearing either at Memory, CPU, or both levels.
+
+Pprof itself runs as part of the the Controller. You can extract the statistics, and visualize them on an interactive website. Furthermore, **you don't need to install this package seperately**. Pprof comes packaged as part of Kubebuilder `controller-runtime` [dependency](https://github.com/kubernetes-sigs/controller-runtime/pull/1943) within the scafolded project.
+
+For further information about the Pprof usage, check out the official [docs](https://github.com/google/pprof).
+
+# How to use Pprof in your Codebase
+
+1. In your `cmd/main.go` file, specify the `PprofBindAddress` URL field as part of Controller Manager's configurations:
+
+    ```golang
+    mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+      ....
+      Scheme:                 scheme,
+      LeaderElection:         enableLeaderElection,
+      // PprofBindAddress is the TCP address that the controller should bind to
+      // for serving pprof. Specify the manager address and the port that should be bind.
+      PprofBindAddress:       "127.0.0.1:8082",
+      ...
+    })
+    ```
+
+2. Reinstall the Controller binary.
+
+    ```bash
+    make install
+    ```
+
+3. Start your controller on your localhost:
+
+    ```bash
+    export ENABLE_WEBHOOKS=false # If you're not using a Webhook. Else, set it to true.
+    make run
+    ```
+
+4. Deploy your custom resource on Kind, or any other Kubernetes cluster. Wait for the pods to be ready.
+
+    ```bash
+    kubectl apply -f batch_v1_cronjob.yaml
+    ```
```

Comment:
we do not need to apply the CR/samples right.
Now, I think is only explain how to access the profile data
PS.: the path for the sample is not accurate. The samples are scaffold under config/samples/

Reply:
@camilamacedo86 I've added this step with the assumption that the end user hasn't created a particular CRD based resource. By not having mentioning this, the reader may get confused. Basically, all the steps that I've are the ones that i ran as a beginner.